neqo-client/src/main.rs

```diff
@@ -452,7 +452,13 @@ fn client(
             max_blocked_streams: args.max_blocked_streams,
         },
     );
-    client.set_qlog(qlog_new(args, hostname)?);
+
+    let qlog_filename = format!(
+        "{}-{}",
+        hostname,
+        client.conn().path().unwrap().remote_cid()
```

Comment:
Can we build an API that wraps this mess.  If the contract is "give me a unique identifier" that is good, but this is poking around in the innards of the connection too much for me.

Reply:
I reduced the ugliness but opened #787 for a proper fix.